File: backend/schemas/payment_proof.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class PaymentProofOut(BaseModel):
    id: str
    tenantId: str
    invoiceId: str
    originalFilename: str
    contentType: str | None
    sizeBytes: int | None
    submittedAt: datetime
    status: str  # pending | approved | applied | rejected
    notes: str | None
    appliedToPaymentId: str | None
    model_config = {"from_attributes": True}


class PaymentProofReviewRequest(BaseModel):
    action: Literal["approve", "reject"]
    notes: str | None = None

File: frontend/src/types/paymentProof.ts
export interface PaymentProof {
  id: string
  tenantId: string
  invoiceId: string
  originalFilename: string
  contentType: string | null
  sizeBytes: number | null
  submittedAt: string
  status: 'pending' | 'approved' | 'applied' | 'rejected'
  notes: string | null
  appliedToPaymentId: string | null
}
